src/cavendish_particle_tracks/_main_widget.py

```diff
@@ -47,15 +48,24 @@ def __init__(
         napari_viewer: napari.Viewer,
         bypass_load_screen: bool = False,
         docking_area: str = "right",
-        shuffling_seed: int = 1,
     ):
         super().__init__()
         self.viewer: napari.Viewer = napari_viewer
         self.bypass_load_screen = bypass_load_screen
         self.docking_area = docking_area
         if self.docking_area != "bottom":
             self.bypass_load_screen = True
-        self.shuffling_seed = shuffling_seed
+        self.shuffling_seed = 1
+
+        if os.getenv("CPT_SHUFFLING_SEED") is not None:
+            try:
+                seed = int(os.environ["CPT_SHUFFLING_SEED"])
+            except ValueError:
+                print(
+                    "Warning: Invalid value for CPT_SHUFFLING_SEED. Using default seed value of 1."
+                )
+            else:
+                self.shuffling_seed = seed
```

Comment:
```suggestion

        self.shuffling_seed = 1
        if os.getenv("CPT_SHUFFLING_SEED"):
            try:
                self.shuffling_seed = int(os.environ["CPT_SHUFFLING_SEED"])
            except ValueError:
                print(
                    "Warning: Invalid value for CPT_SHUFFLING_SEED. Using default seed value of 1."
                )
```

I think this saves you a few lines. I'd separate the `self.shuffling_seed` declaration from the previous lines and move it together with this block too.

You could also pull the whole thing out into a private function, something like:
```python
def _get_shuffling_seed(self, default=1):
    if not os.getenv("CPT_SHUFFLING_SEED")
        return default
    try:
        return int(os.environ["CPT_SHUFFLING_SEED"])
    except ValueError:
        print(f"Warning: Invalid value for CPT_SHUFFLING_SEED. Using default seed value of {default}.")
        return default
```